GVFS/GVFS.Common/Http/HttpRequestor.cs

```diff
@@ -249,10 +303,42 @@ private static string GetSingleHeaderOrEmpty(HttpHeaders headers, string headerN
             IEnumerable<string> values;
             if (headers.TryGetValues(headerName, out values))
             {
-                return values.First();                
+                return values.First();
             }
 
             return string.Empty;
         }
+
+        private string LoadCertificatePassword(string certId, GitProcess git)
+        {
+            if (git.TryGetCertificatePassword(this.Tracer, certId, out var password, out var error))
+            {
+                return password;
+            }
+
+            return null;
+        }
+
+        private X509Certificate2 LoadCertificate(string certId, string certificatePassword)
+        {
+            if (File.Exists(certId))
+            {
+                return new X509Certificate2(certId, certificatePassword);

```

Comment:
Do we need some exception handling in this method since this can throw an exception?

Reply:
Good point, added.